pkg/componentreadiness/tracker/tracker.go

```diff
@@ -0,0 +1,284 @@
+package tracker
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/google/uuid"
+	"github.com/openshift/sippy/pkg/apis/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	testRegressionsDataSet = "ci_analysis_us"
+	testRegressionsTable   = "test_regressions"
+)
+
+type Storage interface {
+	ListCurrentRegressions(release string) ([]api.TestRegression, error)
+	OpenRegression(release string, newRegressedTest api.ComponentReportTestSummary) (*api.TestRegression, error)
+	ReOpenRegression(regressionID string) error
+	CloseRegression(regressionID string, closedAt time.Time) error
+}
+
+// BigQueryStorage is the primary implementation for real world usage, tracking when we detect test
+// regressions opening/closing in the Bigquery test_regressions table.
+type BigQueryStorage struct {
+	client *bigquery.Client
+}
+
+func NewBigQueryStorage(client *bigquery.Client) Storage {
+	return &BigQueryStorage{client: client}
+}
+
+func (bqs *BigQueryStorage) ListCurrentRegressions(release string) ([]api.TestRegression, error) {
+	// List open regressions (no closed date), or those that closed within the last two days. This is to prevent flapping
+	// and return more accurate opened dates when a test is falling in / out of the report.
+	queryString := fmt.Sprintf("SELECT * FROM %s.%s WHERE release = @SampleRelease AND (closed IS NULL or closed > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 2 DAY))",
+		testRegressionsDataSet, testRegressionsTable)
```

Comment:
Stephen has told me (in the past) to tell you it being better to reference bqs.client.DataSet instead of hardcoding is so we can configure the dataset via command options.

Reply:
Ahaaa that's why we have a wrapper client. On it!